keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -910,7 +901,14 @@ class SqlArtifactRepository(
             state = promotionStatus.toLowerCase(),
             deployedAt = deployedAt?.toInstant(ZoneOffset.UTC),
             replacedAt = replacedAt?.toInstant(ZoneOffset.UTC),
-            replacedBy = replacedBy
+            replacedBy = replacedBy,
+            pinned = jooq
+              .fetchExists(
+                ENVIRONMENT_ARTIFACT_PIN,
+                ENVIRONMENT_ARTIFACT_PIN.ENVIRONMENT_UID.eq(deliveryConfig.getUidFor(environmentName))
+                  .and(ENVIRONMENT_ARTIFACT_PIN.ARTIFACT_UID.eq(artifact.uid))
```

Comment:
Here, you're actually repeating those two queries to get the UIDs again. Consider just declaring the variables up top like:
```kotlin
val environmentUid = deliveryConfig.getUidFor(environmentName)
val artifactUid = artifact.uid
```
Then you can reuse them in both places.

Reply:
As far as the nested "exists` query, I think this is fine.

Reply:
These two things just generate the jooq sql, so they're not actually making a new connection to the database. AFAIK making a connection to the database is the problem. But, I could be totally wrong, because I'm new at this